packages/perennial/contracts/multiinvoker/MultiInvoker.sol

```diff
@@ -127,6 +127,24 @@ contract MultiInvoker is IMultiInvoker, UInitializable {
                 (address receiver, IProduct product, UFixed18 amount) = abi.decode(invocation.args, (address, IProduct, UFixed18));
                 withdrawAndUnwrap(receiver, product, amount);
             }
+
+            // Deposit `amount` DSU from `msg.sender` into `vault` on behalf of `account`
+            else if (invocation.action == PerennialAction.VAULT_DEPOSIT) {
```

Comment:
we could actually lock this down so that it has to deposit to `msg.sender`?

Reply:
This is in line with how we do collateral deposit so I was keeping it consistent - I the multiinvoker should give the caller the same flexibility as the original interface as much as possible